pos_self_service_weight_base/static/src/js/Screens/SelfServiceScreen/SelfServiceHomeButton.js

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+odoo.define("pos_self_service_weight_base.SelfServiceHomeButton", function (require) {
+    "use strict";
+
+    const SelfServiceControlButton = require("pos_self_service_weight_base.SelfServiceControlButton");
+    const Registries = require("point_of_sale.Registries");
+    const {useListener} = require("@web/core/utils/hooks");
```

Comment:
@huguesdk i don't understand the difference between these two syntaxes, can you enlighten me ?

Reply:
the second syntax is called [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) and in this case it is equivalent to:

```
    const useListener = require("@web/core/utils/hooks").useListener;
```

it avoids repeating the variable name, but it is even more useful when assigning multiple variables together, like this:

```
    const {useListener, useService} = require("@web/core/utils/hooks");
```

Reply:
thanks !